### PR TITLE
highcharts-theme property

### PR DESCRIPTION
**Pull request by sagarvadodaria**
hctheme
interface HighchartsStatic is missing "theme" property. Typescript
throws error when used the same.Added theme in interface
HighchartsStatic

### CODE REVIEW

**Comment by baltie** [comment]
Sorry for the late response, I'm on vacation. :)

@sagarvadodaria Thanks for your PR. However, I think you are mistaken. I see that in the theme and demo sources, Highcharts themselves set the Highcharts.theme property, but that, I think, is just to avoid polluting the global namespace. In the theme sources, it is straight away used to create a new Highcharts object with that object as the options. There is otherwise nothing in the highcharts sources (highcharts.js, highcharts.more.js, highcharts-3d.js) that makes use of the theme property, so it is not a global theme setting, as far as I can see.

Of course, I might be mistaken, so if you could point towards where in the source this is used or defined or link to documentation, that would be helpful.

**Comment by sagarvadodaria** [comment]
Hello, Wishing you a happy new year.. I was out on vacation as well. :) 

As per the documentation, http://www.highcharts.com/docs/chart-design-and-style/themes 
> "A Highcharts theme is a set of pre-defined options that are applied as default Highcharts options > before each chart is instanciated" ,

so i do want it to be global to have all charts look alike. 

Also setting the theme is done by adding this theme object on the Global Highcharts, so i feel it should be a global property. If there is another way of doing it, I am not sure of it.  I leave it to you to decide how to go about it, but we need some way to stop compilation error in Typescript.

Consider the attached Typescript code and the screenshot which has error. It throws compilation error if I remove below definition in tsd.d.ts. 

interface HighchartsThemeOptions extends HighchartsGlobalOptions {
    legendBackgroundColor?: string;
    background2?: string | HighchartsGradient;
    dataLabelsColor?: string;
    textColor?: string;
    maskColor?: string;
}

interface HighchartsStatic {
    theme: HighchartsThemeOptions;
}


[highcharts.txt](https://github.com/DefinitelyTyped/DefinitelyTyped/files/78083/highcharts.txt)
highcharterror

**Comment by baltie** [comment]
@sagarvadodaria Thanks, happy new year! :)

Thanks for the clarification. Now, if you take a look at the bottom of the page you linked to, you'll notice this call:
`Highcharts.setOptions(Highcharts.theme);`

The parameter could be any highcharts options object, it doesn't have to be `Highcharts.theme`, that's just to avoid to create a new global object in the Highcharts theme code. It is the call to setOptions that's crucial. Take a look at http://api.highcharts.com/highcharts#Highcharts.setOptions, that's how you set global options.

So, you could do the following:
```typescript
var myTheme: HighchartsOptions = {
    chart: { ... }
};

Highcharts.setOptions(myTheme);
```
This would work the same.

If you really want to have a Highcharts.theme object, you can either keep the extension to the HighchartsStatic interface you specified above or set it like this `Highcharts['theme'] = { ... };`

Also, the interface that you have defined, HighchartsThemeOptions, the extra properties there like legendBackgroundColor are only used in the Highcharts demo code and is not part of the Highcharts API. You can only set the legend background color globally like this:
```typescript
Highcharts.setOptions({
    legend: { 
        backgroundColor: '#333333'
    }
});
```

In conclusion, :-1:. The proposed theme property is not part of the API.

**Comment by sagarvadodaria** [comment]
Thanks, i now understand and feel this is not required. Will follow what you said above instead. Thanks again for your time.
